larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.cc

```diff
@@ -387,9 +387,16 @@ float ThreeDReclusteringAlgorithm::GetCheatedFigureOfMerit(const CaloHitList &me
         else
             mainMcParticleMap.insert(std::make_pair(pMainMCParticle, 1));
     }
+    if (mainMcParticleMap.size() == 0)
+    {
+        std::cout << "Cheated figure of merit is incalculable - all hits in a cluster have missing truth matching" << std::endl;
+        throw StatusCodeException(STATUS_CODE_NOT_INITIALIZED);
+    }
     const auto maxSharedHits =
         std::max_element(mainMcParticleMap.begin(), mainMcParticleMap.end(), [](const auto &x, const auto &y) { return x.second < y.second; });
-    const float mainMcParticleFraction = (float)maxSharedHits->second / mergedClusterCaloHitList3D.size();
+    const int totalHits =
+        std::accumulate(mainMcParticleMap.begin(), mainMcParticleMap.end(), 0, [](int tot, const auto &p) { return tot + p.second; });
+    const float mainMcParticleFraction = (float)maxSharedHits->second / (float)totalHits;
```

Comment:
Prefer static_cast<float> here (I'd be inclined to apply it to only one of the variables and let the compiler handle the other one.